packages/discovery-provider/src/api/v1/users.py

```diff
@@ -2670,3 +2680,68 @@ def get(self, id):
         )
         tracks = get_user_tracks_remixed(query_args)
         return success_response(list(map(extend_track, tracks)))
+
+
+USER_PURCHASED_CONTENT_ROUTE = USER_ROUTE + "/purchased"
+
+purchased_response = make_response(
+    "purchased", ns, fields.List(purchasable_content_model)
+)
+
+full_purchased_response = make_full_response(
+    "full_purchased", full_ns, fields.List(purchasable_content_full_model)
+)
+
+
+@ns.route(USER_PURCHASED_CONTENT_ROUTE)
+class UserTracksPurchased(Resource):
+    @ns.doc(
+        id="""Get User Purchased Content""",
+        description="Gets content owned by the user which has been purchased by another user",
+        params={"id": "A User ID"},
+        responses={200: "Success", 400: "Bad request", 500: "Server error"},
+    )
+    @ns.expect(pagination_with_current_user_parser)
+    @ns.marshal_with(purchased_response)
+    def get(self, id):
```

Comment:
Product requirements check: Is this endpoint supposed to be available only to users authorized to see the purchase history of a given user? Or are we okay with being able to see the purchased content of any arbitrary user?
If it's the former, consider using the `@auth_middleware()`/`check_authorized()` pattern